sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_eth.c

```diff
@@ -109,17 +113,16 @@ static inline void _addr_set_broadcast(uint8_t *dst)
     memset(dst, 0xff, ETHERNET_ADDR_LEN);
 }
 
-#define _IPV6_DST_OFFSET    (36)    /* sizeof(ipv6_hdr_t) - 4  */
-
 static inline void _addr_set_multicast(uint8_t *dst, gnrc_pktsnip_t *payload)
 {
     switch (payload->type) {
-#ifdef MODULE_IPV6
+#ifdef MODULE_GNRC_IPV6
         case GNRC_NETTYPE_IPV6:
+            /* https://tools.ietf.org/html/rfc2464#section-7 */
             dst[0] = 0x33;
             dst[1] = 0x33;
-            memcpy(dst + 2, ((uint8_t *)payload->data) + _IPV6_DST_OFFSET, 4);
-            /* TODO change to proper types when gnrc_ipv6_hdr_t got merged */
+            ipv6_hdr_t *ipv6 = (ipv6_hdr_t*)payload->data;
```

Comment:
> first quote is for declarations (and the declaration is correct), second is for casting any pointer to void*, not the other way around...

Yeah, but since this is about style I'm voting for keeping the casting consistent to the declaration. Anyway, happy about how it is now :-)